db/indexes.go

```diff
@@ -363,13 +363,22 @@ func waitForIndex(bucket *base.CouchbaseBucketGoCB, indexName string, queryState
 		}
 		if err == base.ErrViewTimeoutError {
 			base.Infof(base.KeyAll, "Timeout waiting for index %q to be ready for bucket %q - retrying...", base.MD(indexName), base.MD(bucket.GetName()))
+		} else if isRecoverableIndexError(err) {
+			base.Infof(base.KeyAll, "Error waiting for index %q to be ready for bucket %q - retrying...", base.MD(indexName), base.MD(bucket.GetName()))
 		} else {
 			return err
 		}
 	}
 
 }
 
+// Return true if the GSI related errors are recoverable and false
+// otherwise. Index related errors with error code 5000 are considered
+// as recoverable errors.
+func isRecoverableIndexError(err error) bool {
```

Comment:
5000 error codes are not recoverable. They're just the default error code returned by the query service:
https://github.com/couchbase/query/blob/d5cc11221996def63c25a82076ad9a2faa4461f6/errors/errors.go#L60

We should rename this function to be something like `isIndexerError` (and rename the existing function of that name to be more appropriate - `isTransientIndexerError`)

It makes sense for Sync Gateway to keep retrying when we do get these errors at startup, but calling them recoverable is misleading, and is already confusing.

Reply:
Makes sense. This is fixed in latest revision.